### DISCUSSION TITLE
Is it viable to make cloaking as powerful and versatile as filtering?

### DISCUSSION THREAD

**Discussion by msoxzw** [DNSCrypt/dnscrypt-proxy#1950] [Ideas]
Currently, filtering patterns support not only domain name, but also IP address and `CNAME`, so could cloaking patterns support IP address and  `CNAME` as well?

Take `www.wikipedia.org` as an example:

`dig www.wikipedia.org`
```
;; ANSWER SECTION:
www.wikipedia.org.      7161    IN      CNAME   dyna.wikimedia.org.
dyna.wikimedia.org.     46      IN      A       198.35.26.96
```
The IP address of `www.wikipedia.org`, `www.mediawiki.org`, `w.wiki`, etc. is 198.35.26.96. If 198.35.26.96 is very slow or blocked, it can be replaced by another Wikipedia server IP address, e.g. 103.102.166.224

If cloaking patterns support IP address, then `198.35.26.96 103.102.166.224`
If cloaking patterns support `CNAME`, then `dyna.wikimedia.org 103.102.166.224`

Moreover, cloaking, I suggest, could be combined into filtering, i.e. if there is nothing after a pattern, it will be blocked; otherwise, it will be cloaked.

**Comment by jedisct1** [comment]
Cloaking already supports names instead of IP addresses: https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Cloaking#cloaking

**Reply by msoxzw** [reply]
I mean that cloaking patterns support IP address and `CNAME`.
